src/main/java/graphql/normalized/ExecutableNormalizedOperationFactory.java

```diff
@@ -127,19 +167,30 @@ private ExecutableNormalizedOperation createNormalizedQueryImpl(GraphQLSchema gr
 
         ImmutableListMultimap.Builder<Field, ExecutableNormalizedField> fieldToNormalizedField = ImmutableListMultimap.builder();
         ImmutableMap.Builder<ExecutableNormalizedField, MergedField> normalizedFieldToMergedField = ImmutableMap.builder();
+        ImmutableMap.Builder<ExecutableNormalizedField, QueryDirectives> normalizedFieldToQueryDirectives = ImmutableMap.builder();
         ImmutableListMultimap.Builder<FieldCoordinates, ExecutableNormalizedField> coordinatesToNormalizedFields = ImmutableListMultimap.builder();
 
+        BiConsumer<ExecutableNormalizedField, MergedField> captureMergedField = (enf, mergedFld) -> {
+            //QueryDirectivesImpl is a lazy object and only computes itself when asked for
+            QueryDirectives queryDirectives = new QueryDirectivesImpl(mergedFld, graphQLSchema, coercedVariableValues.toMap(), graphQLContext, locale);
+            normalizedFieldToQueryDirectives.put(enf, queryDirectives);
+            normalizedFieldToMergedField.put(enf, mergedFld);
+        };
+
```

Comment:
This conumer is the side effect to create QueryDirectives and also to capture enf -> mergedField